R/plotPValHist.R

```diff
@@ -31,104 +30,146 @@
+        transparency <- 0.3
+    }
+
     if (is.matrix(P.Val)) {
         P.Val <- P.Val %>%
             as.data.frame
     }
 
-    NumSamples <- ncol(P.Val)
-    SampNames <- colnames(P.Val)
+    samples_num  <- ncol(P.Val)
+    sample_names <- colnames(P.Val)
 
     # Set up Tall format
+
     P.Val$GeneID = rownames(P.Val)
     P.Val <-  stats::reshape(data          = P.Val,
                              idvar         = "GeneID",
-                             varying       = SampNames,
-                             v.names       = "Pval",
+                             varying       = sample_names,
+                             v.names       = "pval",
                              direction     = "long",
-                             timevar       = "Levels",
-                             times         = SampNames,
-                             new.row.names = sequence(prod(length(SampNames), nrow(P.Val))))
+                             timevar       = "levels",
+                             times         = sample_names,
+                             new.row.names = sequence(prod(length(sample_names), nrow(P.Val))))
 
     title <- "P-value Histograms"
     plotlist <- list()
-    if (plotType == "canvasXpress") {
+    if (plotType == "canvasxpress") {
+        events <- JS("{'mousemove' : function(o, e, t) {
+                      if (o != null && o != false) {
+                          count = o.y.data[0][0];
+                          bin = o.y.data[0][1];
+                          t.showInfoSpan(e, '<b>Count</b>: ' +
+                          count + '<br/><b>Bin</b>: '+bin);
+                       }; }}")
+
         if (facet) {
-            cx.data <- subset(P.Val, select = Pval)
-            var.annot <- subset(P.Val, select = -c(Pval))
-            plotlist <- canvasXpress::canvasXpress(data                    = cx.data,
-                                                   varAnnot                = var.annot,
-                                                   histogramData           = TRUE,
-                                                   graphType               = "Scatter2D",
-                                                   colors                  = color,
-                                                   title                   = title,
-                                                   xAxisTitle              = "P-value",
-                                                   yAxisTitle              = "Count",
-                                                   hideHistogram           = FALSE,
-                                                   showHistogramDensity    = FALSE,
-                                                   showLegend              = FALSE,
-                                                   segregateVariablesBy    = list("Levels"))
+            cx.data   <- subset(P.Val, select = pval)
```

Comment:
use dplyr **select** and **filter** methods

Reply:
As discussed with Connie, We need to try and use external functions only when required. Will retain this for now. If it has to absolutely be used anywhere, then we can refactor these functions later on to use dplyr equivalent function

Reply:
* There is a task for using dplyr::filter instead of subset here https://3.basecamp.com/3775257/buckets/16965597/todos/3487316376
* Also, tidyr and dplyr function can be used normally as they already imported (we can do everything using base R only but why as long as there are simpler way and they already imported?)